tests/unit/completion/test_models.py

```diff
@@ -269,29 +273,31 @@ def test_url_completion(qtmodeltester, config_stub, web_history, quickmarks,
 
     Verify that:
         - quickmarks, bookmarks, and urls are included
-        - no more than 'web-history-max-items' history entries are included
+        - no more than 'web-history-max-items' items are included (TODO)
```

Comment:
I agree, I was just wondering if this is a usability thing for some people. I'd be happy to scrap it if it means less code though. I tried to implement lazy loading in e1350f3b2. I see `canFetchMore`/`fetchMore` getting called when I load my full history, so hopefully it works as intended? Only benchmarking can tell though.